pkg/util/allocationfilterutil/queryfilters_test.go

```diff
@@ -628,6 +628,58 @@ func TestFiltersFromParamsV1(t *testing.T) {
 				}),
 			},
 		},
+		{
+			name: "single owner",
+			qp: map[string]string{
+				"filterOwners": "nick",
+			},
+			shouldMatch: []kubecost.Allocation{
+				allocGenerator(kubecost.AllocationProperties{
+					Labels: kubecost.AllocationLabels{
+						"testowner": "nick",
+					},
+				}),
+				allocGenerator(kubecost.AllocationProperties{
+					Annotations: kubecost.AllocationAnnotations{
+						"testowner": "nick",
+					},
+				}),
+				allocGenerator(kubecost.AllocationProperties{
+					Labels: kubecost.AllocationLabels{
+						"testowner": "nick",
+					},
+					Annotations: kubecost.AllocationAnnotations{
+						"dontpick": "notnick",
+					},
+				}),
+				allocGenerator(kubecost.AllocationProperties{
+					Labels: kubecost.AllocationLabels{
+						"dontpick": "notnick",
+					},
+					Annotations: kubecost.AllocationAnnotations{
+						"testowner": "nick",
+					},
+				}),
+			},
+			shouldNotMatch: []kubecost.Allocation{
+				allocGenerator(kubecost.AllocationProperties{
+					Labels: kubecost.AllocationLabels{
+						"dontpick": "notnick",
+					},
+					Annotations: kubecost.AllocationAnnotations{
+						"dontpick": "notnick",
+					},
+				}),
+				allocGenerator(kubecost.AllocationProperties{
+					Labels: kubecost.AllocationLabels{
+						"testowner": "notnick",
+					},
+					Annotations: kubecost.AllocationAnnotations{
+						"testowner": "nick",
+					},
+				}),
```

Comment:
@michaelmdresser specifically, this is the test case! Thank you @nickcurie for adding this.

Reply:
If it were just `label[testowner] == "nick" OR annotation[testowner] == "nick"` this would actually pass, but we want it to not pass, based on business requirements. (Whatever we may think about the requirements... is another story.)